qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,231 @@
+            dt_factor_in_microsec = dt_factor_in_sec * 1000000
+            result_unit = "us"
+
+        size = len(experiment_data._data)
+        delays = np.zeros(size, dtype=float)
+        means = np.zeros(size, dtype=float)
+        stddevs = np.zeros(size, dtype=float)
+
+        for i, circ in enumerate(experiment_data._data):
+            delays[i] = circ["metadata"]["delay"] * dt_factor_in_microsec
+            count0 = circ["counts"].get("0", 0)
+            count1 = circ["counts"].get("1", 0)
+            shots = count0 + count1
+            means[i] = count1 / shots
+            stddevs[i] = np.sqrt(means[i] * (1 - means[i]) / shots)
+            # problem for the fitter if one of the std points is
+            # exactly zero
+            if stddevs[i] == 0:
+                stddevs[i] = 1e-4
+
+        if t1_guess is None:
+            t1_guess = np.mean(delays)
+        else:
+            t1_guess = t1_guess * dt_factor_in_microsec
+        if offset_guess is None:
+            offset_guess = means[-1]
+        if amplitude_guess is None:
+            amplitude_guess = means[0] - offset_guess
+        if t1_bounds is None:
+            t1_bounds = [0, np.inf]
+        if amplitude_bounds is None:
+            amplitude_bounds = [0, 1]
+        if offset_bounds is None:
+            offset_bounds = [0, 1]
+
+        fit_out, fit_err, fit_cov, chisq = curve_fit_wrapper(
+            exp_fit_fun,
+            delays,
+            means,
+            stddevs,
+            p0=[amplitude_guess, t1_guess, offset_guess],
+        )
+
+        analysis_result = AnalysisResult(
+            {
+                "value": fit_out[1],
+                "stderr": fit_err[1],
+                "unit": result_unit,
+                "label": "T1",
+                "fit": {
+                    "params": fit_out,
+                    "stderr": fit_err,
+                    "labels": ["amplitude", "T1", "offset"],
+                    "chisq": chisq,
+                    "cov": fit_cov,
+                },
+                "quality": self._fit_quality(fit_out, fit_err, chisq),
+            }
+        )
+
+        return analysis_result, None
+
+    @staticmethod
+    def _fit_quality(fit_out, fit_err, chisq):
+        # pylint: disable = too-many-boolean-expressions
+        if (
+            abs(fit_out[0] - 1.0) < 0.1
+            and abs(fit_out[2]) < 0.1
+            and chisq < 3
+            and (fit_err[0] is None or fit_err[0] < 0.1)
+            and (fit_err[1] is None or fit_err[1] < fit_out[1])
+            and (fit_err[2] is None or fit_err[2] < 0.1)
+        ):
+            return "computer_good"
+        else:
+            return "computer_bad"
+
+
+class T1Experiment(BaseExperiment):
+    """T1 experiment class"""
+
+    __analysis_class__ = T1Analysis
+
+    def __init__(
+        self,
+        qubit: int,
+        delays: Union[List[float], np.array],
+        unit: Optional[str] = "us",
+        experiment_type: Optional[str] = None,
```

Comment:
This arg isn't needed. A custom experiment string should either be defined in the init to be passed with the parent class, or left as none to use the class name.

```suggestion
```

Reply:
This still needs removing